rl/agent/conv_dqn.py

```diff
@@ -40,23 +43,27 @@ def build_hidden_layers(self, model):
                     activation=self.hidden_layers_activation,
                     # border_mode='same',
                     init='lecun_uniform'))
-            num_channels *= 2
-            cols = math.ceil(math.floor(cols  - self.kernel -1) / self.stride[0]) + 1
-            rows = math.ceil(math.floor(rows - self.kernel -1) / self.stride[1]) + 1
+
             for i in range(1, self.num_hidden_layers):
-                if  cols > 5 and rows > 5:
+                num_channels *= 2
+                cols = math.ceil(
+                    math.floor(cols - self.kernel - 1) / self.stride[0]) + 1
+                rows = math.ceil(
+                    math.floor(rows - self.kernel - 1) / self.stride[1]) + 1
+                if cols > 5 and rows > 5:
                     model.add(
                         self.Convolution2D(
-                        num_channels,
-                        self.kernel,
-                        self.kernel,
-                        subsample=self.stride,
-                        activation=self.hidden_layers_activation,
-                        # border_mode='same',
-                        init='lecun_uniform'))
-                    num_channels *= 2
-                    cols = math.ceil(math.floor(cols  - self.kernel -1) / self.stride[0]) + 1
-                    rows = math.ceil(math.floor(rows - self.kernel -1) / self.stride[1]) + 1
+                            num_channels,
+                            self.kernel,
+                            self.kernel,
+                            subsample=self.stride,
+                            activation=self.hidden_layers_activation,
+                            # border_mode='same',
+                            init='lecun_uniform'))
+                else:
+                    # keep growing until it cols/rows bigger? This is disagreeing
+                    # with self.num_hidden_layers
+                    break
```

Comment:
The if move works. For else - break, the cols and row conditions are needed to stop someone adding too many layers, so that cols and rows go to 0. Each layer the number of channels is increased by 2, but the number of cols and rows shrink because the kernel size and stride are greater than 1. 

Two potential solutions: Decrease the kernel size and stride to 1 in the else statement and keep adding layers or 2. Have fewer layers (using the else, break logic) but add an explanation to the python logger.